lib/vector-config-macros/src/configurable.rs

```diff
@@ -167,14 +167,13 @@ fn generate_struct_field(field: &Field<'_>) -> proc_macro2::TokenStream {
     let field_metadata = generate_field_metadata(&field_metadata_ref, field);
 
     let spanned_generate_schema = quote_spanned! {field.span()=>
-        ::vector_config::schema::get_or_generate_schema(schema_gen, #field_metadata_ref.as_subschema())?
+        ::vector_config::schema::get_or_generate_schema(schema_gen, #field_metadata_ref)?
```

Comment:
The `as_subschema` function drops all custom attributes, etc, which is exactly what we _don't_ want to do, as `get_or_generate_schema` is designed to apply that callsite-specific metadata only to the callsite, and not to the underlying `T`.